owrrm/src/main/java/com/facebook/openwifi/rrm/optimizers/tpc/MeasurementBasedApApTPC.java

```diff
@@ -217,7 +219,13 @@ protected static Map<String, List<Integer>> buildRssiMap(
 
 			// At a given AP, if we receive a signal from ap_2, then it gets added to the rssi list for ap_2
 			latestScan.stream()
-				.filter(entry -> (managedBSSIDs.contains(entry.bssid) && UCentralUtils.isChannelInBand(entry.channel, band)))
+				.filter((entry) -> {
+					boolean isManaged = managedBSSIDs.contains(entry.bssid);
```

Comment:
nit: replace with `if (!managedBSSIDs.contains(...)) return false;` - exit early instead of always running the next check